app/api/business/team_business.py

```diff
@@ -0,0 +1,251 @@
+            team = team_service.save(
+                Team(
+                    name='',
+                    status='created',
+                    team_members=[
+                        TeamMember(
+                            user_id=user.id,
+                            is_team_lead=True
+                        )
+                    ]
+                )
+            )
+
+            audit_service.log_audit_event(
+                audit_type=audit_types.create_team,
+                db_object=team,
+                user=current_user.email_address
+            )
+
+            return get_team(team.id)
+
+        created_team = created_teams.pop()
+        return get_team(created_team.id)
+    else:
+        team = completed_teams[0]
+        raise TeamError('You can only be in one team. You\'re already a member of {}.'.format(team.name))
+
+
+def get_team_overview():
+    team_overview = {}
+    user = users.get(current_user.id)
+    completed_teams = team_service.get_teams_for_user(user.id)
+
+    number_of_completed_teams = len(completed_teams)
+    if number_of_completed_teams > 0:
+        team = completed_teams[0]
+        team_overview = team_service.get_team_overview(team.id, user.id)
+
+    organisation = users.get_user_organisation(get_email_domain(user.email_address))
+    team_overview.update(organisation=organisation)
+
+    show_create_team_button = True if number_of_completed_teams == 0 else False
+    team_overview.update(showCreateTeamButton=show_create_team_button)
+
+    return team_overview
+
+
+def get_people_overview():
+    people_overview = {}
+    user = users.get(current_user.id)
+    completed_teams = team_service.get_teams_for_user(user.id)
+    domain = get_email_domain(user.email_address)
+
+    people = users.get_buyer_team_members(user.id, domain)
+    people_overview.update(users=people)
+
+    organisation = users.get_user_organisation(domain)
+    people_overview.update(organisation=organisation)
+
+    show_create_team_button = True if len(completed_teams) == 0 else False
+    people_overview.update(showCreateTeamButton=show_create_team_button)
+
+    return people_overview
+
+
+def get_team(team_id):
+    team = team_service.find(id=team_id).one_or_none()
+    if not team:
+        not_found('No team with id {} found'.format(team_id))
+
+    domain = get_email_domain(current_user.email_address)
+    team = team_service.get_team(team_id)
+
+    if team['teamMembers'] is not None:
+        for user_id, team_member in team['teamMembers'].iteritems():
+            missing_permissions = [permission for permission in permission_types
+                                   if permission not in team_member['permissions']]
+
+            for permission in missing_permissions:
+                team_member['permissions'][permission] = False
+
+    team.update(domain=domain)
+
+    return team
+
+
+def update_team(data):
+    team_id = data.get('id')
+
+    team = team_service.get_team_for_update(data.get('id'))
```

Comment:
This is an insecure direct object reference of sorts - because it is retrieving the team id from the request body, what I can do is use a team id I own in the `/api/2/team/x/update` endpoint, and in the request body, use an id of a team I don't own. I was able to change your team name in RC (sorry!) and could also use this to test which team a user id is a member of (by the "is already a team member of..." error leaking it).

Reply:
To fix, I think this `update_team()` needs to accept the id as a param, and this should be coming from the request URI's id value (i.e. the one which is validated as being owned by the requesting user), not the request body data.

Reply:
fixed